src/NumericalAlgorithms/SphericalHarmonics/IO/StrahlkorperFromFile.cpp

```diff
@@ -0,0 +1,60 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/SphericalHarmonics/Strahlkorper.hpp"
+
+#include <string>
+
+#include "NumericalAlgorithms/SphericalHarmonics/IO/ReadSurfaceYlm.hpp"
+#include "Options/Context.hpp"
+#include "Options/ParseError.hpp"
+#include "Utilities/FileSystem.hpp"
+
+namespace Frame {
+struct Distorted;
+struct Grid;
+struct Inertial;
+}  // namespace Frame
+
+namespace ylm {
+template <typename Frame>
+Strahlkorper<Frame>::Strahlkorper(const size_t l_max,
+                                  const std::string& h5_filename,
+                                  const std::string& subfile_name,
+                                  const double time, const double time_epsilon,
+                                  const bool check_frame,
+                                  const Options::Context& context)
+    : l_max_(l_max), m_max_(l_max), ylm_(l_max, l_max) {
```

Comment:
I think you can just delegate: `: Strahlkorper(l_max, l_max, ylm::read_surface_ylm_single_time(...)) {}`. Not really necessary to do any of the optimizations or extra error handling below. If you want any of those, they can go into `read_surface_ylm_single_time` or into an overload of the prolong/restrict constructor that takes the `other_strahlkorper` by value.